search-app/src/cmr/search/site/data.clj

```diff
@@ -14,30 +14,68 @@
    [clojure.string :as string]
    [cmr.common-app.services.search.query-execution :as query-exec]
    [cmr.common-app.site.data :as common-data]
+   [cmr.common.log :refer :all]
+   [cmr.common.mime-types :as mt]
    [cmr.search.services.query-service :as query-svc]
+   [cmr.search.site.util :as util]
    [cmr.transmit.config :as config]
    [cmr.transmit.metadata-db :as mdb]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Data utility functions
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(defn get-tag-short-name
-  "A utility function used to present a more human-friendly version of a tag."
-  [tag]
-  (case tag
-    "gov.nasa.eosdis" "EOSDIS"))
+(defmulti get-providers
+  "Get the providers, based on contextual data.
 
-(defn collection-data
-  "Get the collection data associated with a provider and tag."
+  The `:cli` variant uses a special constructed context (see
+  `static.StaticContext`).
+
+  The default variant is the original, designed to work with the regular
+  request context which contains the state of a running CMR."
+  :execution-context)
+
+(defmethod get-providers :cli
+  [context]
+  (let [providers-url (format "%sproviders"
+                              (config/application-public-root-url :ingest))]
+    (util/endpoint-get providers-url {:accept mt/json})))
```

Comment:
Why are we wrapping clj-http?

Reply:
We're actually not just wrapping clj-http, but performing a set of other functions as well. This is so we don't have to repeatedly do those, too.